plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -248,6 +244,28 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
     return fetch(url, requestInit);
   }
 
+  private isServiceAccountAuthentication(
+    authProvider: string,
+    clusterDetails: ClusterDetails,
+  ) {
+    return (
+      authProvider === 'serviceAccount' &&
+      !clusterDetails.authMetadata.serviceAccountToken &&
+      fs.pathExistsSync(Config.SERVICEACCOUNT_CA_PATH)
+    );
+  }
+
+  private isTokenOrClientCertificateAuthentication(
+    authProvider: string,
+    credential: KubernetesCredential,
+  ) {
+    return (
+      credential.type === 'bearer token' ||
+      credential.type === 'x509 client certificate' ||
+      authProvider === 'localKubectlProxy'
+    );
+  }
```

Comment:
If you like my comment above, then it makes sense for this to become something like:

```suggestion
  private isCredentialMissing(
    authProvider: string,
    credential: KubernetesCredential,
  ) {
    return (
      authProvider !== 'localKubectlProxy' &&
      credential.type === 'anonymous'  
    );
  }
```